dask_mongo/core.py

```diff
@@ -146,17 +131,16 @@ def read_mongo(
             )
         )
 
-    partitions = [
-        fetch_mongo(
-            connection_args,
-            database,
-            collection,
+    common_args = (connection_args, database, collection, match)
```

Comment:
As is, I think we can have the same Dask collection name even when users specify different values for `chunksize`

```suggestion
    common_args = (connection_args, database, collection, chunksize, match)
```